engine/resource/src/resource_archive.cpp

```diff
@@ -53,6 +54,11 @@ namespace dmResourceArchive
         m_HashOffset = dmEndian::ToHost((uint32_t)sizeof(ArchiveIndex));
     }
 
+    void RegisterResourceDecryption(FDecryptResource decrypt_resource) {
+        assert(g_ResourceDecryption == 0);
+        g_ResourceDecryption = decrypt_resource;
+    }
+
```

Comment:
This does not match the implementation in the build pipeline where you register multiple encryptions